From 4ebad9c40fa8c5a815ca549b88fce59191b2fd08 Mon Sep 17 00:00:00 2001
Message-Id: <4ebad9c40fa8c5a815ca549b88fce59191b2fd08.1650636481.git.keller@fs-net.de>
In-Reply-To: <8876d3464f7abe86d96420dca69dd3317368c208.1650636480.git.keller@fs-net.de>
References: <8876d3464f7abe86d96420dca69dd3317368c208.1650636480.git.keller@fs-net.de>
From: bings <bings@codeaurora.org>
Date: Mon, 6 Apr 2020 12:57:37 +0800
Subject: [PATCH 25/56] qcacld-2.0: Driver changes to accumulate kernel changes
 related to FILS

With kernel version 4.19, a new structure
cfg80211_fils_resp_params is introduced for fils parameters in
the existing structure cfg80211_connect_resp_params. The
functions hdd_populate_fils_params and hdd_connect_done use
cfg80211_fils_resp_params as argument for modifying the fils
parameters. This causes compilation issues with 4.19 kernel.

Add changes to accumulate cfg80211_fils_resp_params structure
addition from 4.19 kernel.

Change-Id: I6bac3a38a3d065d33be73bbd88bc974f3155070d
CRs-Fixed: 2657551
---
 CORE/HDD/src/wlan_hdd_main.c | 45 +++++++++++++++++++++++++++++++++++++++-----
 1 file changed, 40 insertions(+), 5 deletions(-)

diff --git a/CORE/HDD/src/wlan_hdd_main.c b/CORE/HDD/src/wlan_hdd_main.c
index cd56ec8..969b55c 100644
--- a/CORE/HDD/src/wlan_hdd_main.c
+++ b/CORE/HDD/src/wlan_hdd_main.c
@@ -13339,8 +13339,11 @@ static void hdd_connect_bss(struct net_device *dev, const u8 *bssid,
 #endif
 
 #ifdef WLAN_FEATURE_FILS_SK
-#ifdef CFG80211_CONNECT_DONE
-#ifdef CFG80211_FILS_SK_OFFLOAD_SUPPORT
+#if (defined(CFG80211_CONNECT_DONE) || \
+	(LINUX_VERSION_CODE >= KERNEL_VERSION(4, 12, 0))) && \
+	(LINUX_VERSION_CODE < KERNEL_VERSION(4, 19, 0))
+#if defined(CFG80211_FILS_SK_OFFLOAD_SUPPORT) || \
+		 (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 12, 0))
 /**
  * hdd_populate_fils_params() - Populate FILS keys to connect response
  * @fils_params: connect response to supplicant
@@ -13378,6 +13381,39 @@ static inline void hdd_populate_fils_params(struct cfg80211_connect_resp_params
 { }
 #endif /* CFG80211_FILS_SK_OFFLOAD_SUPPORT */
 
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 19, 0))
+/**
+ * hdd_populate_fils_params() - Populate FILS keys to connect response
+ * @fils_params: connect response to supplicant
+ * @fils_kek: FILS kek
+ * @fils_kek_len: FILS kek length
+ * @pmk: FILS PMK
+ * @pmk_len: FILS PMK length
+ * @pmkid: PMKID
+ * @fils_seq_num: FILS Seq number
+ *
+ * Return: None
+ */
+static void hdd_populate_fils_params(struct cfg80211_connect_resp_params
+				     *fils_params, const uint8_t *fils_kek,
+				     size_t fils_kek_len, const uint8_t *pmk,
+				     size_t pmk_len, const uint8_t *pmkid,
+				     uint16_t fils_seq_num)
+{
+	/* Increament seq number to be used for next FILS */
+	fils_params->fils.erp_next_seq_num = fils_seq_num + 1;
+	fils_params->fils.update_erp_next_seq_num = true;
+	fils_params->fils.kek = fils_kek;
+	fils_params->fils.kek_len = fils_kek_len;
+	fils_params->fils.pmk = pmk;
+	fils_params->fils.pmk_len = pmk_len;
+	fils_params->fils.pmkid = pmkid;
+}
+#endif /* CFG80211_CONNECT_DONE */
+
+#if defined(CFG80211_CONNECT_DONE) || \
+	(LINUX_VERSION_CODE >= KERNEL_VERSION(4, 12, 0))
+
 /**
  * hdd_connect_done() - Wrapper API to call cfg80211_connect_done
  * @dev: network device
@@ -13430,9 +13466,8 @@ static void hdd_connect_done(struct net_device *dev, const u8 *bssid,
                      roam_fils_params->fils_pmkid,
                      roam_info->fils_seq_num);
     }
-    hddLog(LOG1, "FILS indicate connect status %d seq no %d",
-          fils_params.status,
-          fils_params.fils_erp_next_seq_num);
+    hddLog(LOG1, "FILS indicate connect status %d",
+          fils_params.status);
 
     cfg80211_connect_done(dev, &fils_params, gfp);
 
-- 
2.14.5

